win32ss/user/user32/windows/window.c

```diff
@@ -924,52 +924,40 @@ GetAltTabInfoW(HWND hwnd,
     return NtUserGetAltTabInfo(hwnd,iItem,pati,pszItemText,cchItemText,FALSE);
 }
 
-
-/*
- * @implemented
- */
+/* @implemented */
 HWND WINAPI
-GetAncestor(HWND hwnd, UINT gaFlags)
+GetAncestor(_In_ HWND hwnd, _In_ UINT uType)
 {
-    HWND Ret = NULL;
-    PWND Ancestor, Wnd;
-
-    Wnd = ValidateHwnd(hwnd);
-    if (!Wnd)
+    PWND pWnd = ValidateHwnd(hwnd);
+    if (!pWnd || pWnd == GetThreadDesktopWnd())
         return NULL;
 
-    _SEH2_TRY
+    if (uType == GA_PARENT)
     {
-        Ancestor = NULL;
-        switch (gaFlags)
-        {
-            case GA_PARENT:
-                if (Wnd->spwndParent != NULL)
-                    Ancestor = DesktopPtrToUser(Wnd->spwndParent);
-                break;
+        /* Optimized for speed */
```

Comment:
This comment could be moved just before the `if (uType == GA_PARENT)` case.
`/* Special handling optimized for speed */`

Reply:
Fixed.